e2etest/layer2_test.go

```diff
@@ -217,8 +217,8 @@ var _ = ginkgo.Describe("L2", func() {
 
 		ip := firstIPFromRange(*ipRange)
 		svc1, err := jig1.CreateLoadBalancerService(loadBalancerCreateTimeout, func(svc *corev1.Service) {
-			svc.Spec.Ports[0].TargetPort = intstr.FromInt(82)
-			svc.Spec.Ports[0].Port = 82
+			svc.Spec.Ports[0].TargetPort = intstr.FromInt(int(servicePodPort))
```

Comment:
can we change the type of servicePodPort to int to avoid these extra casts? Would it introduce other extra casts?

Reply:
we will still have to cast the Port which is an int32. and also i wanted to keep it as uint to avoid negative port.

Reply:
Yes but that's a parameter the user is setting. The double cast is just anti-esthetic and I don't think there's much value in having that enforced.

Reply:
ok so we can cast just the port